src/commands/deploy.rs

```diff
@@ -750,3 +747,12 @@ fn print_available_routes(
         }
     }
 }
+
+// Check if the token has expired.
+// If the expiration is None, assume the token has not expired
+fn has_expired(login_connection: &LoginConnection) -> bool {
+    match &login_connection.expiration {
+        Some(expiration) => Utc::now() > DateTime::parse_from_rfc3339(&expiration).unwrap(),
```

Comment:
The `unwrap` here is "crash the program if the parse fails", which is not what you want - sure, it _shouldn't_ happen, but it's not a program-stopping disaster.  I'd suggest falling back to the safe option: if we can't parse the expiry string, assume it's expired - it's more fiddly nesting but totally worth it.  E.g.

```suggestion
        Some(expiration) => match DateTime::parse_from_rfc3339(&expiration) {
            Ok(time) => Utc::now() > time,
            Err(_) => true,
        }
```

Reply:
done